app/mailers/concerns/mailer_defaults_configurable_concern.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module MailerDefaultsConfigurableConcern
+  extend ActiveSupport::Concern
+
+  class_methods do
+    # Save original defaults before they're modified
+    def save_original_defaults
+      @original_default_from ||= self.default[:from]
+      @original_default_host ||= Rails.application.routes.default_url_options[:host]
+      @original_asset_host ||= asset_host
+    end
+
+    # Resets mailer settings to their original values
+    def reset_original_defaults
+      default from: original_default_from, reply_to: original_default_from
+      default_url_options[:host] = original_default_host
+      Rails.application.routes.default_url_options[:host] = original_default_host
+      self.asset_host = original_asset_host
+    end
+
+    def original_default_from = @original_default_from
+    def original_default_host = @original_default_host
+    def original_asset_host = @original_asset_host
+  end
+
+  included do
+    before_action -> { self.class.save_original_defaults }
+    before_action :set_currents_for_legacy
+    after_action -> { self.class.reset_original_defaults }
+
+    def configure_defaults_for_user(user)
+      return if user.nil? || !user.is_a?(User) # not for super-admins
```

Comment:
```suggestion
      return if !user.is_a?(User) # not for super-admins
```

`NilClass != UserClass` j'adore ruby ... a moins que tu voulais être explicite ?

Reply:
non j'ai rajouté la seconde condition après, corrigé